Sources/MapboxNavigation/CarPlayManager.swift

```diff
@@ -699,24 +709,31 @@ extension CarPlayManager: CPMapTemplateDelegate {
                   return
               }
         
-        guard let routeResponse = routeChoice.routeResponseFromUserInfo,
-              let routeOptions = routeResponse.options as? RouteOptions else {
-                  preconditionFailure("CPRouteChoice should contain `RouteResponseUserInfo` struct.")
-              }
+        guard let indexedRouteResponse = routeChoice.indexedRouteResponseUserInfo?.indexedRouteResponse else {
+            preconditionFailure("CPRouteChoice should contain `IndexedRouteResponseUserInfo` struct.")
+        }
 
         mapTemplate.hideTripPreviews()
         
         let desiredSimulationMode: SimulationMode = simulatesLocations ? .always : .inTunnels
         
+        var navigationServiceWithRouteOptions: (() -> NavigationService?)? = nil
+        if case let .route(routeOptions) = indexedRouteResponse.routeResponse.options {
+            navigationServiceWithRouteOptions = {
+                self.delegate?.carPlayManager(self,
+                                              navigationServiceFor: indexedRouteResponse.routeResponse,
+                                              routeIndex: indexedRouteResponse.routeIndex,
+                                              routeOptions: routeOptions,
+                                              desiredSimulationMode: desiredSimulationMode)
+            }
+        }
+        
         let navigationService = self.navigationService ??
         delegate?.carPlayManager(self,
-                                 navigationServiceFor: routeResponse.response,
-                                 routeIndex: routeResponse.routeIndex,
-                                 routeOptions: routeOptions,
+                                 navigationServiceFor: indexedRouteResponse,
                                  desiredSimulationMode: desiredSimulationMode) ??
-        MapboxNavigationService(routeResponse: routeResponse.response,
-                                routeIndex: routeResponse.routeIndex,
-                                routeOptions: routeOptions,
+        navigationServiceWithRouteOptions?() ??
```

Comment:
> Another downside - warning message about calling a deprecated method.

A private protocol can [redefine this deprecated method](https://github.com/mapbox/mapbox-directions-swift/blob/3e0d1ac37aca0489965028e4521623531a02cb7d/Sources/MapboxDirections/MapMatching/MatchOptions.swift#L138-L140) as a non-deprecated method, so that internally we can [cast and call](https://github.com/mapbox/mapbox-directions-swift/blob/3e0d1ac37aca0489965028e4521623531a02cb7d/Sources/MapboxDirections/MapMatching/MatchOptions.swift#L110) the method, ignoring that public deprecation.

Reply:
Unfortunately this trick won't work with protocols. Protocol extension cannot have inheritance clause, and we also cannot define public protocol extending a private one.